lib/Utils.js

```diff
@@ -64,11 +64,13 @@ class Utils {
     }
 
```

Comment:
You are right, but I left those params untouched as is it would break compatibility with the current build of typeorm. 
See [here](https://github.com/typeorm/typeorm/blob/master/src/driver/sap/SapDriver.ts#L213)

Reply:
Understood.

One concern is: the connection will fail if someone uses ```serverNode``` instead of ```hostName``` and ```port``` (this is possible if parameters can be sent directly). To prevent from being overwritten with ```undefined``` value, it's better to move ```...params``` to the end of the dictionary, and add two lines of code to get hostName/port from serverNode.

```javascript
    const hostName = params.hostName? params.hostName : params.serverNode? params.serverNode.split(":")[0] : undefined;
    const port = params.port? params.port : params.serverNode? params.serverNode.split(":")[1] : undefined;

    return isHANAClient ? {
      HOST: hostName,
      PORT: port,
      UID: params.userName,
      PWD: params.password,
      ...params
    } : {
      host: hostName,
      port: port,
      user: params.userName,
      password: params.password,
      ...params
    };
```
What do you think?

Reply:
Sure! Updated as you suggested.